azure-pipelines.yml

```diff
@@ -0,0 +1,55 @@
+trigger:
+  branches:
+    include:
+    - master
+  tags:
+    include:
+    - v*
+
+strategy:
+  matrix:
+    linux:
+      imageName: 'ubuntu-latest'
+    mac:
+      imageName: 'macos-latest'
+    windows:
+      imageName: 'windows-latest'
+
+pool:
+  vmImage: $(imageName)
+
+steps:
+
+- task: NodeTool@0
+  inputs:
+    # We have had strange failures on Windows with Node.js v10, v12 doesn't seem
+    # to present the same issue
+    #
+    #     [main 2021-02-17T12:35:24.272Z] [File Watcher (node.js)] Error: ENOENT: no such file or directory, stat 'c:\Users\VssAdministrator\AppData\Roaming\Code\User'
+    #     [...]
+    #       Extension Test Suite
+    #
+    #         √ Sample test
+    #
+    #       1 passing (58ms)
+    #
+    #     Exit code:   3765269347
+    #     Done
+    versionSpec: '12.x'
+  displayName: 'Install Node.js'
+
+- bash: |
+    /usr/bin/Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+    echo ">>> Started xvfb"
+  displayName: Start xvfb
+  condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
+
+- bash: |
+    echo ">>> Compile $(Build.Repository.Name)"
+    npm install && npm run compile
+    echo ">>> Compiled $(Build.Repository.Name)"
+    echo ">>> Run integration test"
+    npm install && npm run compile && npm run test
```

Comment:
Now we're installing and compiling twice before running the test (before, there was a `cd sample`) in between.